test/unit/Search.test.js

```diff
@@ -148,14 +148,20 @@ describe('Search', () => {
     let triggerSubmitSpy;
     let submitSearchRequestSpy;
     let mock;
+    let contextRoutesPushed;
 
-    before(() => {
+    beforeEach(() => {
```

Comment:
`beforeEach` runs for each test in the block. `before` runs once for the block it is in. In my earlier suggestion, I meant adding a separate `afterEach` hook with `reset`, which resets the data from the spy. `restore` gets rid of the spy/mock entirely. Now, the `afterEach` hook could also manage resetting `contextRoutesPushed` to an empty array. Sorry if my comment was confusing.

Reply:
yeah, I just thought it would be easier to create/tear-down all of the mocks/spies in between each test, but I had forgotten that `reset` is different from `restore` honestly.

I think both approaches work but yours is definitely a bit cleaner.